agenda/domain/src/main/java/com/gkp/agenda/domain/AgendaRepository.kt

```diff
@@ -1,23 +1,19 @@
 package com.gkp.agenda.domain
 
 import com.gkp.agenda.domain.model.AgendaItem
-import com.gkp.core.domain.util.TaskyResult
 import kotlinx.coroutines.flow.Flow
 
 interface AgendaRepository {
-    fun fetchAgendaItems(time: Long = System.currentTimeMillis()) : Flow<TaskyResult<List<AgendaItem>>>
+    fun getAgendaItemsForDate(dateLong: Long): Flow<List<AgendaItem>>
 
-    fun addAgendaItem(agendaItem: AgendaItem)
+    suspend fun fetchAgendaItems()
 
-    fun updateAgendaItem(agendaItem: AgendaItem)
+    suspend fun addAgendaItem(agendaItem: AgendaItem)
 
-    fun deleteAgendaItem(agendaItem: AgendaItem)
+    suspend fun updateAgendaItem(agendaItem: AgendaItem)
 
-    fun fetchAgendaItemTask(id: String) : Flow<TaskyResult<AgendaItem.Task>>
-
-    fun fetchAgendaItemReminder(id: String) : Flow<TaskyResult<AgendaItem.Reminder>>
-
-    fun fetchAgendaItemEvent(id: String) : Flow<TaskyResult<AgendaItem.Event>>
+    suspend fun deleteAgendaItem(agendaItem: AgendaItem)
+    suspend fun getAgendaItemForId(id: String): AgendaItem
```

Comment:
Could it not be possible that the AgendaItem does not exist for the ID provided? 🤔